packages/core/test-support/helpers/contains-helpers.js

```diff
@@ -27,3 +27,48 @@ export function buttonContains(text) {
 export function linkContains(text) {
   return findByContains('a', text);
 }
+
+/**
+ * This function allows us to keep using the jquery notation for :contains without using jquery
+ *
+ * Need quotes around content with parentheses -- e.g. class:contains("(content)")
+ *
+ *
+ * Example:
+ * Jquery: $(.class-thing:contains(blah) .metric .test:contains(other))
+ *
+ * Non-Jquery Equivalent: [
+ *    ...[
+ *      ...document.querySelectorAll(
+ *        (.class-thing)
+ *      )].find(el => el.innerText.trim().includes('blah'))
+ *    .querySelectorAll('.metric .test')
+ *  ]
+ *  .find(el => el.innerText.trim().includes('other'))
+ *
+ * findContains: findContains(.class-thing:contains(blah) .metric .test:contains(other))
+ *
+ * @param {String} selector - jquery or native style selector
+ * @param {HTMLElement} [baseElement] - optional parameter to use as base element to select off of
+ * @returns {HTMLElement} - Element based on text it (or its parent) contains
+ */
+export function findContains(selector, baseElement) {
```

Comment:
Problem: Consider the case
```html
<div class="a">dog</div>
<div class="b">dog</div>
```
and we do a `findContains(':contains(dog).b')`.  

We would expect to find the second division, but it is lost because the iteration is done on only the first find of the contains.

Solution: Do the recursion with a `findContainsAll` function that takes an array of selectors instead of a singleton, then make `findContains` a special case of that function.